scripts/linters/pre_commit_linter.py

```diff
@@ -509,23 +536,27 @@ def main(args=None):
     # (ie. might parallelize on their own)
 
     # Concurrency limit: 25.
-    concurrent_task_utils.execute_tasks(tasks_custom, custom_semaphore)
+    if 'custom' in linter_type_used:
+        concurrent_task_utils.execute_tasks(tasks_custom, custom_semaphore)
     # Concurrency limit: 2.
-    concurrent_task_utils.execute_tasks(
-        tasks_third_party, third_party_semaphore)
+    if 'third_party' in linter_type_used:
+        concurrent_task_utils.execute_tasks(
+            tasks_third_party, third_party_semaphore)
 
     all_messages = []
 
     # Prepare semaphore for locking mechanism.
     semaphore = threading.Semaphore(1)
 
-    for task in tasks_custom:
-        semaphore.acquire()
-        _get_task_output(all_messages, task, semaphore)
+    if 'custom' in linter_type_used:
+        for task in tasks_custom:
+            semaphore.acquire()
```

Comment:
We used semaphore because we needed a locking mechanism, sot that lint errors of one file do not get printed under another file, and semaphore is the better in handling these situations rather than simple locking mechanism. Because in simple locking we have a risk of running in a deadlock situation.

We can't use this in TaskThread class because there we have semaphore with concurrency limit of 25 or 2 or any other variable. But here we need concurrency limit as 1 because here we need to access one thread at a time.

Maybe we can use a context manager, but I am not sure. semaphore is release in `_get_task_output` method.